Modules/getpath.py

```diff
@@ -0,0 +1,679 @@
+                if EXEC_PREFIX:
+                    exec_prefix = EXEC_PREFIX
+                    if not isdir(joinpath(exec_prefix, PLATSTDLIB_LANDMARK)):
+                        warn("Could not find platform dependent libraries <exec_prefix>")
+                else:
+                    warn("Could not find platform dependent libraries <exec_prefix>")
+            else:
+                exec_prefix = new_exec_prefix
+
+
+    # Fallback: assume exec_prefix == prefix
+    if not exec_prefix:
+        exec_prefix = prefix
+
+
+    if not prefix or not exec_prefix:
+        warn("Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]")
+
+
+if not base_prefix:
+    base_prefix = prefix
+
+if not base_exec_prefix:
+    base_exec_prefix = exec_prefix
+
+
+# If we haven't set [plat]stdlib_dir already, set them now
+if not stdlib_dir:
+    if prefix:
+        stdlib_dir = joinpath(prefix, STDLIB_SUBDIR)
+    else:
+        stdlib_dir = ""
+
+if not platstdlib_dir:
+    if exec_prefix:
+        platstdlib_dir = joinpath(exec_prefix, PLATSTDLIB_LANDMARK)
+    else:
+        platstdlib_dir = ""
+
+
+# For a venv, update the main prefix/exec_prefix but leave the base ones unchanged
+# XXX: We currently do not update prefix here, but it happens in site.py
+#if venv_prefix:
+#    prefix = exec_prefix = venv_prefix
+
+
+# ******************************************************************************
+# UPDATE pythonpath (sys.path)
+# ******************************************************************************
+
+if py_setpath:
+    # If Py_SetPath was called then it overrides any existing search path
+    config['module_search_paths'] = py_setpath.split(DELIM)
+    config['module_search_paths_set'] = 1
+
+elif not pythonpath:
+    # If pythonpath was already set, we leave it alone.
+    # This won't matter in normal use, but if an embedded host is trying to
+    # recalculate paths while running then we do not want to change it.
+    pythonpath = []
+
+    # First add entries from the process environment
+    if use_environment and ENV_PYTHONPATH:
+        for p in ENV_PYTHONPATH.split(DELIM):
+            pythonpath.append(abspath(p))
+
+    # Then add the default zip file
+    if os_name == "nt":
+        # QUIRK: Windows uses the executable_directory rather than the prefix
+        pythonpath.append(joinpath(executable_dir, ZIP_LANDMARK))
+    elif build_prefix or venv_prefix:
+        # QUIRK: POSIX uses the default prefix when in the build directory
+        # or a venv
+        pythonpath.append(joinpath(PREFIX, ZIP_LANDMARK))
+    else:
+        pythonpath.append(joinpath(prefix, ZIP_LANDMARK))
+
+    if os_name == "nt" and use_environment:
+        # QUIRK: Windows also lists paths in the registry. Paths are stored
+        # as the default value of each subkey of
+        # {HKCU,HKLM}\Software\Python\PythonCore\{winver}\PythonPath
+        # where winver is sys.winver (typically '3.x' or '3.x-32')
+        try:
+            import winreg
```

Comment:
Why not pass `winreg` in like we do with `config`?  Then the import machinery isn't involved via the interpreter.

Reply:
You wouldn't have to trigger `__import__()`, instead doing something more focused.  I figured there'd be something like `PyImport_ImportFrozenModule()` for builtin modules, but apparently not.  You could do something like `_PyImport_BootstrapImp()`, especially since it's the same situation.

Reply:
Made this change now, and added a test that uses a mock module. Using `PyImport_ImportModule` works fine, though it's tempting to expose `_imp.create_builtin` (but not tempting enough!)